math/src/main/java/com/powsybl/math/timeseries/TimeSeriesTable.java

```diff
@@ -0,0 +1,597 @@
+
+    public class Correlation {
+
+        private final String timeSeriesName1;
+        private final String timeSeriesName2;
+        private final double coefficient;
+
+        public Correlation(String timeSeriesName1, String timeSeriesName2, double coefficient) {
+            this.timeSeriesName1 = Objects.requireNonNull(timeSeriesName1);
+            this.timeSeriesName2 = Objects.requireNonNull(timeSeriesName2);
+            this.coefficient = coefficient;
+        }
+
+        public String getTimeSeriesName1() {
+            return timeSeriesName1;
+        }
+
+        public String getTimeSeriesName2() {
+            return timeSeriesName2;
+        }
+
+        public double getCoefficient() {
+            return coefficient;
+        }
+
+        @Override
+        public String toString() {
+            return "Correlation(timeSeriesName1=" + timeSeriesName1 + ", timeSeriesName2=" + timeSeriesName2 +
+                    ", coefficient=" + coefficient + ")";
+        }
+    }
+
+    private class TimeSeriesNameMap {
+
+        private final List<String> indexToName = new ArrayList<>();
+
+        private final TObjectIntMap<String> nameToIndex = new TObjectIntHashMap<>();
+
+        int add(String name) {
+            int i = indexToName.size();
+            indexToName.add(name);
+            nameToIndex.put(name, i);
+            return i;
+        }
+
+        String getName(int index) {
+            if (index < 0 || index >= indexToName.size()) {
+                throw new IllegalArgumentException("Time series at index " + index + " not found");
+            }
+            return indexToName.get(index);
+        }
+
+        int getIndex(String name) {
+            Objects.requireNonNull(name);
+            if (!nameToIndex.containsKey(name)) {
+                throw new IllegalArgumentException("Time series '" + name + "' not found");
+            }
+            return nameToIndex.get(name);
+        }
+
+        int size() {
+            return indexToName.size();
+        }
+    }
+
+    private int fromVersion;
+
+    private int toVersion;
+
+    private List<TimeSeriesMetadata> timeSeriesMetadata;
+
+    private TimeSeriesIndex tableIndex;
+
+    private final TIntArrayList timeSeriesIndexDoubleOrString = new TIntArrayList(); // global index to typed index
+
+    private final TimeSeriesNameMap doubleTimeSeriesNames = new TimeSeriesNameMap();
+
+    private final TimeSeriesNameMap stringTimeSeriesNames = new TimeSeriesNameMap();
+
+    private DoubleBuffer doubleBuffer;
+
+    private CompactStringBuffer stringBuffer;
+
+    // statistics
+    private double[] means;
+
+    private double[] stdDevs;
+
+    public TimeSeriesTable(int fromVersion, int toVersion) {
+        TimeSeriesIndex.checkVersion(fromVersion);
+        TimeSeriesIndex.checkVersion(toVersion);
+        if (toVersion < fromVersion) {
+            throw new TimeSeriesException("To version is expected to be greater tha from version");
```

Comment:
Change the error message: you can use the variable name in the message instead of "To version" and "from Version". You also should display the values. And you forgot a "n" to "than":

throw new TimeSeriesException("toVersion (" + toVersion" + ") is expected to be greater than fromVersion (" + fromVersion +")");

Reply:
Done